sync.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+const (
+	adminGroupMembersURI = "https://www.googleapis.com/admin/directory/v1/groups/%s/members"
+	adminUserURI         = "https://www.googleapis.com/admin/directory/v1/users/%s?customFieldMask=keys&projection=custom"
+)
+
+// Google group members
+type GoogleMemberList struct {
+	Members []GoogleMember `json:"members"`
+}
+
+type GoogleMember struct {
+	Email string `json:"email"`
+}
+
+// Google admin user
+type GoogleAdminUser struct {
+	CustomSchemas GoogleCustomSchema `json:"customSchemas"`
+}
+
+type GoogleCustomSchema struct {
+	Keys GoogleKeys `json:"keys"`
+}
+
+type GoogleKeys struct {
+	SSH string `json:"ssh"`
+}
+
+type Group struct {
+	Name string   `json:"name"`
+	Keys []string `json:"keys"`
+}
+
+func authmap(adminClient *http.Client, groups []string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var rtnGroups []Group
+		for _, g := range groups {
+			var memList GoogleMemberList
+			group := Group{Name: g, Keys: []string{}}
+			req, _ := http.NewRequest(http.MethodGet, fmt.Sprintf(adminGroupMembersURI, g), nil)
+			resp, _ := adminClient.Do(req)
```

Comment:
Close resp body so that tcp connections are reused

Reply:
Do I still need to read it all like so? 

https://github.com/utilitywarehouse/ssh-key-manager/blob/master/main.go#L73-L76

Reply:
That's how I usually do it. `io.Copy(ioutil.Discard, resp.Body)` might be redundant though.